configure.ac

```
@@ -724,14 +726,22 @@ popdef([host])dnl
 OCAML_WITH_NONEXECSTACK_NOTE
 OCAML_ASM_SIZE_TYPE_DIRECTIVES
 
+## Find vendor of the C compiler
+OCAML_CC_VENDOR
+
+DEP_CC_PREFIX=''
 AS_CASE([$host],
   [sparc-sun-solaris*],
     [DEP_CC="false"],
   [*-pc-windows],
-    [AC_CHECK_TOOLS(
-      [DEP_CC],
-      [$DEP_CC gcc cc x86_64-w64-mingw32-gcc i686-w64-mingw32-gcc],
-      [false])],
+    [AS_CASE([$ocaml_cc_vendor],
+      [msvc-*-clang-*],
+      [DEP_CC="$CC"
+      DEP_CC_PREFIX='/clang:'],
+      [AC_CHECK_TOOLS(
+        [DEP_CC],
+        [$DEP_CC gcc cc x86_64-w64-mingw32-gcc i686-w64-mingw32-gcc],
```

```suggestion
        [$DEP_CC x86_64-w64-mingw32-gcc i686-w64-mingw32-gcc gcc cc],
```

I don't think this changes a lot but if we have includes guarded under `#ifdef _WIN32` they will be caught with mingw-w64, but not with Cygwin's gcc.

---

Hah - there is one place (in st_stubs.c) where we `#include` a local header guarded with `_WIN32`... but it's a good idea regardless!

---

Of course, perhaps the better hardening would simply be to add `-D_WIN32` as well when computing the deps?

---

> Hah - there is one place (in st_stubs.c) where we `#include` a local header guarded with `_WIN32`... but it's a good idea regardless!

I did spend a few weeks digging in that file ;)